lib/worker/init.ts

```diff
@@ -12,24 +12,61 @@ require('../common/reset-search-paths');
 // Import common settings.
 require('@electron/internal/common/init');
 
-// Export node bindings to global.
-const { makeRequireFunction } = __non_webpack_require__('internal/modules/cjs/helpers') // eslint-disable-line
-global.module = new Module('electron/js2c/worker_init');
-global.require = makeRequireFunction(global.module);
-
-// Set the __filename to the path of html file if it is file: protocol.
-if (self.location.protocol === 'file:') {
-  const pathname = process.platform === 'win32' && self.location.pathname[0] === '/' ? self.location.pathname.substr(1) : self.location.pathname;
-  global.__filename = path.normalize(decodeURIComponent(pathname));
-  global.__dirname = path.dirname(global.__filename);
-
-  // Set module's filename so relative require can work as expected.
-  global.module.filename = global.__filename;
-
-  // Also search for module under the html file.
-  global.module.paths = Module._nodeModulePaths(global.__dirname);
+// Process command line arguments.
+const { hasSwitch, getSwitchValue } = process._linkedBinding('electron_common_command_line');
+
+const parseOption = function (name: string) {
+  return hasSwitch(name) ? getSwitchValue(name) : null;
+};
+
+const nodeIntegration = hasSwitch('node-integration-in-worker');
+const preloadScript = parseOption('preload-in-worker');
+
+if (nodeIntegration) {
+  // Export node bindings to global.
```

Comment:
This file shares lots of code with `lib/renderer/init.ts`, can you move the shared code into a utility function? Like what `windowSetup` and `webViewInit` do.